src/eu/deltasource/internship/ecosystem/ecosystems/Ecosystem.java

```diff
@@ -0,0 +1,184 @@
+package eu.deltasource.internship.ecosystem.ecosystems;
+
+import eu.deltasource.internship.ecosystem.animals.Animal;
+import eu.deltasource.internship.ecosystem.animals.Carnivore;
+import eu.deltasource.internship.ecosystem.animals.Herbivore;
+import eu.deltasource.internship.ecosystem.enums.LivingTypes;
+import eu.deltasource.internship.ecosystem.utilities.RandomNumberGeneratorRealInput;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class Ecosystem {
+
+    public static List<Carnivore> carnivores = new ArrayList<>();
+    public static List<Herbivore> herbivores = new ArrayList<>();
+
+    public static void init(List<Animal> animalCollection) {
+
+        for (int i = 0; i < animalCollection.size(); i++) {
+            if (animalCollection.get(i) instanceof Carnivore) {
+                carnivores.add((Carnivore) animalCollection.get(i));
+            } else {
+                herbivores.add((Herbivore) animalCollection.get(i));
+            }
+        }
+    }
+
+    public static void run(int iterations) {
+
+        reproduce();
+        attack();
+        age();
+        ageCheck();
+        increaseStarvation();
+        starvationCheck();
+        cleanDeadAnimals();
+        sortLists();
+        System.out.println("Iteration : " + iterations);
+        System.out.println("Herbivores : " + herbivores.size());
+        System.out.println("Carnivore : " + carnivores.size() + "\n");
+
+
+    }
+
+    /**
+     * Sorts the list of carnivores and herbivores.
+     */
+    private static void sortLists() {
+        Collections.sort(carnivores);
+        Collections.sort(herbivores);
+    }
+
+    /**
+     * Increase the hunger level of the carnivores by their hunger rate modifier
+     */
+    private static void increaseStarvation() {
+        for (int i = 0; i < carnivores.size(); i++) {
+            carnivores.get(i).setHungerLevel(carnivores.get(i).getHungerLevel() + carnivores.get(i)
```

Comment:
Instead of calling setHungerLevel, I would rather let the animal "get hungrier" itself ;-)